components/Markets/MarketOrdersCard.tsx

```diff
@@ -0,0 +1,62 @@
+import { ActionIcon, Group, Loader, Stack, Tabs, Text } from '@mantine/core';
+import { useWallet } from '@solana/wallet-adapter-react';
+import { IconRefresh } from '@tabler/icons-react';
+import { useOpenbookMarket } from '@/contexts/OpenbookMarketContext';
+import {
+  isEmptyOrder,
+  _isOpenOrder,
+} from '@/lib/openbook';
+import { OpenbookMarket } from '@/lib/types';
+import { OpenOrdersTab } from '../Orders/OpenOrdersTab';
+import { UnsettledOrdersTab } from '@/components/Orders/UnsettledOrdersTab';
+
+export function MarketOrdersCard({
+  market,
+}: {
+  market: OpenbookMarket;
+}) {
+  const wallet = useWallet();
+  const { fetchOpenOrders, orders } = useOpenbookMarket();
+
+  if (!wallet || !orders) return <></>;
+
+  return !wallet || !orders ? (
+    <Group justify="center" w="100%" h="100%">
+      <Loader />
+    </Group>
+  ) : (
+    <>
+      <Stack gap={2}>
+        <Group justify="space-between" align="flex-start">
+          <Text fw="bolder" size="xl">
+            Orders
+          </Text>
+          <ActionIcon
+            variant="subtle"
+            // @ts-ignore
+            onClick={() => fetchOpenOrders(wallet.publicKey)}
+          >
+            <IconRefresh />
+          </ActionIcon>
+        </Group>
+        <Stack justify="start" align="start" />
+      </Stack>
+      {orders.length > 0 ?
+      <Tabs defaultValue="open">
+        <Tabs.List>
+          <Tabs.Tab value="open">Open</Tabs.Tab>
+          <Tabs.Tab value="unsettled">Unsettled</Tabs.Tab>
+        </Tabs.List>
+        <Tabs.Panel value="open">
+          <OpenOrdersTab orders={orders.filter((order) => _isOpenOrder(order, market))} />
```

Comment:
we should try to memoize values like this with something like 
```
const openOrders = useMemo(() => orders.filter((order) => _isOpenOrder(order, market), [orders.length])
```
because a filter function is a bit expensive to redo on each render. Ideally, we only re calc this when the base array length changes.

Reply:
Fixed in incoming commit.